csharp/src/SeedLang/IEngine.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2021 The Aha001 Team.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Collections.Generic;
+using SeedLang.Common;
+
+namespace SeedLang {
+  // The interface of the SeedLang engine.
+  public interface IEngine {
+    // A collection that contains all binary operators.
+    IEnumerable<string> BinaryOperators { get; }
```

Comment:
Since we have re-defined the dependency graph, I think there is no need for client code (Unity UI) to get the list of supported binary operators from IEngine directly:

Yes, Unity UI will need to know all the supported operators but it can also talk to SeedLang.Block to get the list. And, since SeedLang.Block will have its PartialParser and FullParser inside the assembly boundary, its PartialParser and FullParser must have the knowledge of supported binary operators (e.g., they can get that knowledge from the ANTLR generated code based on grammars/seed_block.g4). Hence, Unity UI can surely get the operator list from SeedLang.Block, right?

Given this, I think it's better not to expose this interface from IEngine.

My original request of this interface was based on my previous idea that SeedLang.Block depends on IEngine. The request is obsolete now since the dependency does not exist any more.

Reply:
Sure. I will remove this interface now. And I'm going to add the Block parser in the next two or three PRs. We can see if it's easy to get the operators from SeedBlock parser.